src/main/java/org/commcare/formplayer/application/FormSubmissionController.java

```diff
@@ -91,7 +98,8 @@ public SubmitResponseBean submitForm(@RequestBody SubmitRequestBean submitReques
                 stepFactory.makeStep("processFormXml", this::processFormXml, PROCESSED_XML),
                 stepFactory.makeStep("updateVolatility", this::updateVolatility),
                 stepFactory.makeStep("performSync", this::performSync),
-                stepFactory.makeStep("doEndOfFormNav", this::doEndOfFormNav, PROCESSED_STACK)
+                stepFactory.makeStep("doEndOfFormNav", this::doEndOfFormNav, PROCESSED_STACK),
+                stepFactory.makeStep("clearCaches", this::clearCaches)
```

Comment:
I am concerned that this is going to cause [this issue](https://github.com/dimagi/formplayer/blob/autoLaunchTest/src/test/java/org/commcare/formplayer/tests/MultiSelectCaseClaimTest.java#L111) for session navigation requests after EOF navigation to a screen that comes after a case search result screen. Curious if it's possible to clear the cache is at the start of a new menu session instead ?

Reply:
The stack for EOF nav does separate queries which should not match the cache keys for the existing searches. You can see an example here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/tests/test_suite_registry_search.py#L208-L221

Obviously I'll need to test that but I don't think it should be an issue.

Reply:
got it, these points make perfect sense. thanks!